### PR TITLE
docs: Revamp the "job templates" section; Asset handling

### PR DESCRIPTION
**Pull request by okurz**
The section "Using job templates to automate jobs creation" was part of
"GettingStarted" for a long time but did not really help beginners so this
commit moves it into the Users Guide.

The wording is updated to mention "job groups" as the place where the
test scenarios are actually defined. The necessary test definition steps are
updated to mention "job groups" in the correct place.

"Priorities" were previously incorrectly described with "test suites" and were
therefore moved to the section about "job groups" where they actually appear
in the web UI.

The order of sub-sections is adjusted according to the suggested order of
definition in the introductory section description.

The different ways how the test scenarios can be configured within the job
groups are described.

### CODE REVIEW

**Comment by okurz** [comment]
Follow-up to https://github.com/os-autoinst/openQA/pull/2127 . If #2127 is merged first the diff gets easier :)

File: docs/UsersGuide.asciidoc [hunk 1]
```diff
@@ -157,65 +398,163 @@ the same page of the table.
...
+
+==== Workflow for creating or updating needles ====
+
+1. In case a new needles should be created, add the corresponding +assert_screen+s to
+   your test.
+2. Start the test with the +assert_screen+s which are supposed to fail.
+3. Select "Pause on +assert_screen+ failure and confirm.
+4. Wait until the test has paused. There is a button to skip the current timeout to speed
+   this up.
+5. A button for acessing the needle editor should occour. It may take a few seconds till
```
Comment (Martchus) [inline docs/UsersGuide.asciidoc]: occour -> occur

Comment (okurz) [reply docs/UsersGuide.asciidoc]: I'm just moving the section with your typo, you know that, right? ;) Fixed, updated

Comment (Martchus) [reply docs/UsersGuide.asciidoc]: I know

**Pull request merged**